### PR TITLE
Compensate for changed GetFeatureInfo response

### LINKED ISSUE #198: Spatial selection is broken on the raster portals
On the raster portals, the code that updates the download URL seems to be broken. The spatial sub-select component of the download URL doesn't get updated, and the following error shows up in the console.

```
Uncaught TypeError: bbox is undefined
    getRasterNativeProj https://data.pacificclimate.org/portal/js/pdp_raster_map.js:130
    onBoxChange https://data.pacificclimate.org/portal/js/pdp_controls.js:562
    canada_ex_app https://data.pacificclimate.org/portal/js/canada_ex_app.js:85
    jQuery 3
    getNCWMSLayerCapabilities https://data.pacificclimate.org/portal/js/data-services.js:50
    jQuery 6
    getNCWMSLayerCapabilities https://data.pacificclimate.org/portal/js/data-services.js:44
    canada_ex_app https://data.pacificclimate.org/portal/js/canada_ex_app.js:81
    triggerEvent https://data.pacificclimate.org/portal/js/OL/OpenLayers-2.13.1.js:138
    generateMenuTree https://data.pacificclimate.org/portal/js/pdp_controls.js:199
    jQuery 8
    generateMenuTree https://data.pacificclimate.org/portal/js/pdp_controls.js:197
    each jQuery
    generateMenuTree https://data.pacificclimate.org/portal/js/pdp_controls.js:185
    generateMenuTree https://data.pacificclimate.org/portal/js/pdp_controls.js:189
    each jQuery
    generateMenuTree https://data.pacificclimate.org/portal/js/pdp_controls.js:185
    generateMenuTree https://data.pacificclimate.org/portal/js/pdp_controls.js:189
    each jQuery
    generateMenuTree https://data.pacificclimate.org/portal/js/pdp_controls.js:185
    generateMenuTree https://data.pacificclimate.org/portal/js/pdp_controls.js:189
    each jQuery
    generateMenuTree https://data.pacificclimate.org/portal/js/pdp_controls.js:185
    getRasterAccordionMenu https://data.pacificclimate.org/portal/js/pdp_controls.js:213
    jQuery 6
    getRasterAccordionMenuData https://data.pacificclimate.org/portal/js/data-services.js:24
    getRasterAccordionMenu https://data.pacificclimate.org/portal/js/pdp_controls.js:212
    getRasterControls https://data.pacificclimate.org/portal/js/pdp_controls.js:256
    canada_ex_app https://data.pacificclimate.org/portal/js/canada_ex_app.js:56
    jQuery 11
pdp_raster_map.js:130:15
    getRasterNativeProj https://data.pacificclimate.org/portal/js/pdp_raster_map.js:130
    onBoxChange https://data.pacificclimate.org/portal/js/pdp_controls.js:562
    canada_ex_app https://data.pacificclimate.org/portal/js/canada_ex_app.js:85
    jQuery 3
    getNCWMSLayerCapabilities https://data.pacificclimate.org/portal/js/data-services.js:50
    jQuery 6
    getNCWMSLayerCapabilities https://data.pacificclimate.org/portal/js/data-services.js:44
    canada_ex_app https://data.pacificclimate.org/portal/js/canada_ex_app.js:81
    triggerEvent https://data.pacificclimate.org/portal/js/OL/OpenLayers-2.13.1.js:138
    generateMenuTree https://data.pacificclimate.org/portal/js/pdp_controls.js:199
    jQuery 8
    generateMenuTree https://data.pacificclimate.org/portal/js/pdp_controls.js:197
    each jQuery
    generateMenuTree https://data.pacificclimate.org/portal/js/pdp_controls.js:185
    generateMenuTree https://data.pacificclimate.org/portal/js/pdp_controls.js:189
    each jQuery
    generateMenuTree https://data.pacificclimate.org/portal/js/pdp_controls.js:185
    generateMenuTree https://data.pacificclimate.org/portal/js/pdp_controls.js:189
    each jQuery
    generateMenuTree https://data.pacificclimate.org/portal/js/pdp_controls.js:185
    generateMenuTree https://data.pacificclimate.org/portal/js/pdp_controls.js:189
    each jQuery
    generateMenuTree https://data.pacificclimate.org/portal/js/pdp_controls.js:185
    getRasterAccordionMenu https://data.pacificclimate.org/portal/js/pdp_controls.js:213
    jQuery 6
    getRasterAccordionMenuData https://data.pacificclimate.org/portal/js/data-services.js:24
    getRasterAccordionMenu https://data.pacificclimate.org/portal/js/pdp_controls.js:212
    getRasterControls https://data.pacificclimate.org/portal/js/pdp_controls.js:256
    canada_ex_app https://data.pacificclimate.org/portal/js/canada_ex_app.js:56
    jQuery 
```

@rod-glover and @corviday is this related to the ncWMS upgrade?

Steps to reproduce:

1. Go to the PRISM map [here](https://data.pacificclimate.org/portal/bc_prism/map/)
2. Observe the download URL. Looks something like this: https://data.pacificclimate.org/data/bc_prism/pr_mClimMean_PRISM_historical_19710101-20001231.nc.nc?climatology_bounds,pr[][0:1680][0:3241]&
3. Select a rectangle on the map
4. Observe the download URL. It's the same. One would expect the spatial component to change... e.g. https://data.pacificclimate.org/data/bc_prism/pr_mClimMean_PRISM_historical_19710101-20001231.nc.nc?climatology_bounds,pr[][500:600][100:150]&

### PR DESCRIPTION
**Pull request by rod-glover**
Resolves #198 

This PR also updates the `local-dev` Docker infrastructure to allow a fully functional PDP application to be run locally.

### CODE REVIEW

File: pdp/static/js/utils.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      return i;
+    }
+    const k = Math.floor(i + f * d);
+    return v < a[k] ? binarySearch(a, v, i, k) : binarySearch(a, v, k, j);
+  }
+
+
+  function nearestIndex(a, v) {
+    // Given an array `a` in ascending order, and a value `v`, return the index
+    // `k` such that `a[k] <= v < a[k+1]` if `a[0] <= v < a[n-1]`, and
```
Comment (rod-glover) [inline pdp/static/js/utils.js:27]: Wrong description. Fix.

**Review by corviday — COMMENTED** [review]
The code here looks good and well thought out. I want to spin it up and test it real quick before approving.

**Comment by corviday** [comment]
I have been unable to get the local dev test setup working. (Error message `502 Bad Gateway` from nginx). I'm going to see if I can get it running on one of our docker-servers, but I may need to get help from @rod-glover or @matthewbenstead for testing.

**Comment by corviday** [comment]
Okay, finally, several days later, I have gotten tests running! Sorry for the delay. 

Everything seems to work well on the `bc_prism`, `hydro_model_archive`, and `pcds`, `hydro_stn_archive`. and `hydro_stn_cmip5` portals.

When I load the `gridded_observations`, `downscaled_gcms`, `downscaled_gcms_archive`, `hydro_model_out`,  portals, I get a popup that says: `There was an unhandleable problem fetching some metadata for the current layer. Unfortunately, this prevents some of the controls from functioning properly. Please try a page reload and if the problem persists, please file a bug report.` The console shows a CORS error. Perhaps I still don't have the docker test setup configured correctly, do you see this error on your end?

Drawing a selection area on these portals results in an error (`Uncaught TypeError: bbox is undefined`) and does not update the download link.
-------------
Weirdly, the `hydro_model_archive` has this issue on the *live* portal.

**Comment by rod-glover** [comment]
@corviday , I don't get the CORS error, nor the error message you quote. Instead, I get correct-looking successful requests to the backend, and this error: "Cannot resolve selection to data grid. Please zoom in and select only within the data region." 

I think the error you see is possibly due to setup. Can you send me a copy of the relevant error messages from the Network and/or Console tabs?

I think the error I see is a real bug, and I am investigating. It would be worthwhile getting your setup to the point where you can see this "better" error.

**Comment by rod-glover** [comment]
I think I have found the root of the real error. The latitude array is, for the `daily_gridded` datasets, in descending order. For most other datasets, it is in ascending order.

**Comment by rod-glover** [comment]
@corviday , a demo version is up at https://services.pacificclimate.org/dev/portal/bc_prism/map/ etc. I believe everything is working correctly. Would you mind casting your eye over it again? 

The code has been modified slightly to handle ascending and descending spatial (lat, lon) dimension arrays. You can look at if you like or skip it.

**Review by corviday — APPROVED** [review]
I tested each portal, and downloaded data from a couple of them. It all looks good to me!

**Pull request merged**